test/fhirtypes/ElementDefinition.test.ts

```diff
@@ -319,6 +319,53 @@ describe('ElementDefinition', () => {
       expect(observation.elements[codeIdx + 5].id).toBe('Observation.subject');
     });
 
+    it('should add children from the Structure Definition when they exist', () => {
+      const numOriginalElements = observation.elements.length;
+      const component = observation.elements.find(e => e.path === 'Observation.component');
+      component.slicing = {
+        ordered: false,
+        rules: 'open',
+        discriminator: [{ type: 'value', path: 'code' }]
+      };
```

Comment:
BTW, `ElementDefinition` has a `sliceIt` function.  So you could have done:
```ts
component.sliceIt('value', 'code');
```

Reply:
I wasn't sure if it was better to try to avoid depending on other functions if I could, but I can change that if you want the compactness.

Reply:
Oh, I see.  Personally, I think it would be fine since `sliceIt` isn't terribly complicated and is well-tested elsewhere (and you're already using `slice`) -- but I understand your point, so I'm fine with it as-is.